Source/Globals.cpp

```diff
@@ -57,9 +61,47 @@ namespace android::global
     }
 
     void Initialize(JavaVM* javaVM, jobject context)
+    {
+        Initialize(javaVM, context, nullptr);
+    }
+
+    void Initialize(JavaVM* javaVM, jobject context, jobject assetManager)
     {
         g_javaVM = javaVM;
-        g_appContext = GetEnvForCurrentThread()->NewGlobalRef(android::content::Context{context}.getApplicationContext());
+        JNIEnv* env = GetEnvForCurrentThread();
+
+        if (g_appContext != nullptr)
+        {
+            env->DeleteGlobalRef(g_appContext);
+            g_appContext = nullptr;
+        }
+
+        if (context != nullptr)
+        {
+            android::content::Context contextWrapper{context};
+            const auto applicationContextWrapper = contextWrapper.getApplicationContext();
+            jobject applicationContext = applicationContextWrapper;
```

Comment:
The point of the wrapper objects is to hide the details of interacting with Java to make it easier. Saving all the wrappers in a separate object and using the jobjects directly makes the code awkward. If we actually must check for the existence of the applicationContext object, we can add an implicit boolean operator to avoid having to save the wrapper in a different variable.